src/main/webapp/app/exercises/shared/submission/submission.service.ts

```diff
@@ -42,8 +42,10 @@ export class SubmissionService {
                 tap((res) =>
                     res.body!.forEach((submission) => {
                         // reconnect results to submissions
-                        if (getLatestSubmissionResult(submission)) {
-                            getLatestSubmissionResult(submission)!.submission = submission;
+                        const tmp = getLatestSubmissionResult(submission);
```

Comment:
I don’t really like this code as it is duplicated. Can this not be handled by the same function in submission.model.ts ? Also we should not call the variable tmp

Reply:
Is now fixed and refactored